contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,964 @@
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /**
+     * @dev Check that the Aerodrome pool price is within the expected
+     *      parameters.
+     *      This function ignores whether the strategy contract has liquidity
+     *      position in the pool.
+     */
+    function _checkForExpectedPoolPrice() internal {
+        require(
+            allowedWethShareStart != 0 && 
+            allowedWethShareEnd != 0
+        , "Weth share interval not set");
+        
+        uint160 _currentPrice = getPoolX96Price();
+
+        /**
+         * First check we are in expected tick range
+         *
+         * We revert even though price being equal to the lower tick would still
+         * count being within lower tick for the purpose of Sugar.estimateAmount calls
+         */
+        if (
+            _currentPrice <= sqrtRatioX96TickLower ||
+            _currentPrice >= sqrtRatioX96TickHigher
+        ) {
+            revert OutsideExpectedTickRange(getCurrentTradingTick());
+        }
+
+        /**
+         * If estimateAmount1 call fails it could be due to _currentPrice being really
+         * close to a tick and amount1 too big to compute.
+         *
+         * If token addresses were reversed estimateAmount0 would be required here
+         */
+        uint256 _normalizedWethAmount = 1 ether;
+        uint256 _correspondingOethAmount = helper.estimateAmount1(
+            _normalizedWethAmount,
+            address(0), // no need to pass pool address when current price is specified
+            _currentPrice,
+            lowerTick,
+            upperTick
+        );
+
+        // 18 decimal number expressed weth tick share
+        uint256 _wethSharePct = _normalizedWethAmount.divPrecisely(
+            _normalizedWethAmount + _correspondingOethAmount
+        );
+
+        if (
+            _wethSharePct < allowedWethShareStart ||
+            _wethSharePct > allowedWethShareEnd
+        ) {
+            revert PoolRebalanceOutOfBounds(
+                _wethSharePct,
+                allowedWethShareStart,
+                allowedWethShareEnd
+            );
+        }
+        emit PoolRebalanced(_wethSharePct);
+    }
+
+    /**
+     * Burns any OETHb tokens remaining on the strategy contract
+     */
+    function _burnOethbOnTheContract() internal {
+        uint256 _oethbBalance = IERC20(OETHb).balanceOf(address(this));
+        if (_oethbBalance > 0) {
+            IVault(vaultAddress).burnForStrategy(_oethbBalance);
+        }
+    }
+
+    /// @dev this function assumes there are no uncollected tokens in the clPool owned by the.
+    ///      strategy contract. For that reason any liquidity withdrawals also collect the tokens.
+    function _updateUnderlyingAssets() internal {
+        if (tokenId == 0) {
+            underlyingAssets = 0;
+            emit UnderlyingAssetsUpdated(underlyingAssets);
+            return;
+        }
+
+        uint128 _liquidity = _getLiquidity();
```

Comment:
⚪ Would be nice to short-circuit if `liquidity == 0`(just like the tokenId check above)

Reply:
If we add this, we would only add it for code readability. When liquidity is 0 then the helper function returns 0 for both token amounts. I've added this as a comment for more clarity: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/8097f4556f104b8e0f6f2164d959d68bdb8958d0